app/models/mail_handler.rb

```diff
@@ -161,8 +161,8 @@ def receive_issue
     end
 
     issue = WorkPackage.new(author: user, project: project)
-    issue.safe_attributes = issue_attributes_from_keywords(issue)
-    issue.safe_attributes = { 'custom_field_values' => custom_field_values_from_keywords(issue) }
+    issue.attributes = issue_attributes_from_keywords(issue)
```

Comment:
The `safe_attributes=` on WorkPackages is overwriting the one from `Redmine::SafeAttributes`. Restoring the calls to it will not work because the method is not working after your removal of the module. The method calls `delete_unsafe_attributes` which is defined in `Redmine::SafeAttributes`. So the `safe_attributes=` method on `WorkPackage` has to be removed and we need to find an alternate place for the logic within it.  

There is another method within `WorkPackage` called `update_by!`/`update_by` which could be a place for the business logic. One notable thing is though, that said method was circumventing the business logic defined in `safe_attributes=` the whole time. Not quite sure where that leaves us. As far as I can tell, the `update_by` method is used in the more important places and there haven't been any bug reports regarding behaviour on that paths. So we might be able to delete the method altogether without moving the logic.

Another deeper look at what is in `safe_attributes=` will tell you that. 

If we decide not to delete (parts) of the method, another place to put the logic would be the `UpdateWorkPackageService` by the way, which already exists.